proto/multipoolermanagerservice.proto

```diff
@@ -0,0 +1,113 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package multipoolermanager;
+
+import "multipoolermanagerdata.proto";
+
+option go_package = "github.com/multigres/multigres/go/pb/multipoolermanager";
+
+// MultiPoolerManager provides management APIs for PostgreSQL connection poolers
+service MultiPoolerManager {
+  //
+  // PG Replication API wrappers. 
+  //
+
+  // WaitForLSN waits for PostgreSQL server to reach a specific LSN position
+  rpc WaitForLSN(multipoolermanagerdata.WaitForLSNRequest) returns (multipoolermanagerdata.WaitForLSNResponse);
+
+  // SetReadOnly makes the PostgreSQL instance read-only
+  rpc SetReadOnly(multipoolermanagerdata.SetReadOnlyRequest) returns (multipoolermanagerdata.SetReadOnlyResponse);
+
+  // PromoteStandby PostgreSQL standby server to primary
+  // this will call pg_promote() in the underlying instance.
+  rpc PromoteStandby(multipoolermanagerdata.PromoteStandbyRequest)
+    returns (multipoolermanagerdata.PromoteStandbyResponse);
+
+  // GetPrimaryLSN gets the current leader LSN position
+  rpc GetPrimaryLSN(multipoolermanagerdata.GetPrimaryLSNRequest)
+    returns (multipoolermanagerdata.GetPrimaryLSNResponse);
+
+  // IsReadOnly checks if PostgreSQL instance is in read-only mode
+  rpc IsReadOnly(multipoolermanagerdata.IsReadOnlyRequest)
+    returns (multipoolermanagerdata.IsReadOnlyResponse);
+
+  // SetStandbyPrimaryConnInfo sets the primary connection info for a standby server
+  rpc SetStandbyPrimaryConnInfo(multipoolermanagerdata.SetStandbyPrimaryConnInfoRequest)
+    returns (multipoolermanagerdata.SetStandbyPrimaryConnInfoResponse);
+
+  // StartStandbyReplication starts WAL replay on standby (calls pg_wal_replay_resume)
+  rpc StartStandbyReplication(multipoolermanagerdata.StartReplicationRequest)
+    returns (multipoolermanagerdata.StartReplicationResponse);
+
+  // StopStandbyReplication stops WAL replay on standby (calls pg_wal_replay_pause)
+  rpc StopStandbyReplication(multipoolermanagerdata.StopStandbyReplicationRequest)
+    returns (multipoolermanagerdata.StopStandbyReplicationResponse);
+
+  // StandbyReplicationStatus gets the current replication status of the standby
+  rpc StandbyReplicationStatus(multipoolermanagerdata.StandbyReplicationStatusRequest)
+    returns (multipoolermanagerdata.StandbyReplicationStatusResponse);
+
+  // ResetStandbyReplication resets the standby's connection to its primary by clearing primary_conninfo,
+  // reloading PostgreSQL configuration, and terminating the WAL receiver process. This effectively
+  // disconnects the replica from the primary and prevents it from acknowledging commits, making it
+  // unavailable for synchronous replication until reconfigured.
+  rpc ResetStandbyReplication(multipoolermanagerdata.ResetStandbyReplicationRequest)
```

Comment:
In my mind this is what we will be calling when recruiting. The response contains LSN and latest transaction timestamp.